ivc/tests/add.rs

```diff
@@ -467,36 +470,152 @@ pub fn heavy_test_simple_add() {
     );
 
     ////////////////////////////////////////////////////////////////////////////
-    // Witness step 1 & 2
+    // Witness step 1 (APP + IVC)
     ////////////////////////////////////////////////////////////////////////////
 
-    // Starting building witnesses. We start with the application witnesses. It
-    // will be used after that to build the witness for the IVC
-    let mut _app_witness_one: AppWitnessBuildEnv = {
-        let mut env = WitnessBuilderEnv::create();
-
-        for _i in 0..domain_size {
-            let a: Fp = Fp::rand(&mut rng);
-            let b: Fp = Fp::rand(&mut rng);
-            env.write_column(AdditionColumn::A, &a);
-            env.write_column(AdditionColumn::B, &b);
-            interpreter_simple_add(&mut env);
-            env.next_row();
-        }
-        env
+    println!("Building witness step 1");
+
+    type IVCWitnessBuilderEnvRaw<LT> =
+        WitnessBuilderEnv<Fp, IVCColumn, N_WIT_IVC, N_WIT_IVC, 0, N_BLOCKS, LT>;
+    type LT = IVCLookupTable<Fq>;
+
+    let mut ivc_witness_env_0 = IVCWitnessBuilderEnvRaw::<LT>::create();
+
+    let mut app_witness_one: AppWitnessBuilderEnv = WitnessBuilderEnv::create();
+
+    let empty_lookups_app = BTreeMap::new();
+    let empty_lookups_ivc = BTreeMap::new();
+
+    // Witness one
+    for _i in 0..domain_size {
+        let a: Fp = Fp::rand(&mut rng);
+        let b: Fp = Fp::rand(&mut rng);
+        app_witness_one.write_column(AdditionColumn::A, &a);
+        app_witness_one.write_column(AdditionColumn::B, &b);
+        app_witness_one.write_column(AdditionColumn::C, &(a * a * b));
+        interpreter_simple_add(&mut app_witness_one);
+        app_witness_one.next_row();
+    }
+
+    let proof_inputs_one = app_witness_one.get_proof_inputs(domain_size, empty_lookups_app.clone());
+    assert!(proof_inputs_one.evaluations.len() == 3);
+
+    ivc_witness_env_0.set_fixed_selectors(ivc_fixed_selectors.clone());
+    ivc_circuit_base_case::<Fp, _, N_COL_TOTAL, N_CHALS>(&mut ivc_witness_env_0, domain_size);
+    let ivc_proof_inputs_0 =
+        ivc_witness_env_0.get_proof_inputs(domain_size, empty_lookups_ivc.clone());
+    assert!(ivc_proof_inputs_0.evaluations.len() == N_WIT_IVC);
+    for i in 0..10 {
+        assert!(
+            ivc_proof_inputs_0.evaluations[0][i] == Fp::zero(),
+            "Iteration column row #{i:?} must be zero"
+        );
+    }
+
+    // FIXME this merely concatenates two witnesses. Most likely, we
+    // want to intersperse them in a smarter way later. Our witness is
+    // Relation || dynamic.
+    let joint_witness_one: Vec<_> = proof_inputs_one
+        .evaluations
+        .clone()
+        .into_iter()
+        .chain(ivc_proof_inputs_0.evaluations.clone())
+        .collect();
+
+    for i in 0..10 {
```

Comment:
Why?

Reply:
Just a sanity check that I used for debugging. Can be removed in the future.